### PR TITLE
Privacy analysis

### PR DESCRIPTION
**Pull request by martinthomson**
This is a little fiddly, but I think that the general conclusion makes sense:

* Clients advertise support, but rarely need to send TRAIN packets.
* ...except for real-time or other bidirectional exchange patterns.
* ...maybe.
* Servers use opportunities that clients offer, but probably don't care much about uplink, except for real-time/bidirectional exchanges.

Privacy is important, so I'd like to get this in a reasonable shape before we ship a -00.  It's probably the last thing to do though.

### CODE REVIEW

**Review by huitema — COMMENTED** [review]
In short, you are saying that clients don't care because the actual rate can be observed, and service don't care because their access is shared between multiple connections.

I don't think that you can separate client/server so neatly. Suppose a video conference application. Both client and server are trying to send and receive data, both need to tune the coding rate of video images -- and if we move to telepresence both could use very large bandwidth.

File: draft-thomson-train-protocol.md [hunk 1]
```diff
@@ -456,7 +456,25 @@ might benefit from understanding rate limits might consume rate limit signals
 when clients enable the feature. However, servers have to operate correctly even
 when clients act to protect the privacy of their service configuration.
 
+## Conditional Enablement
 
+As a signal that is passively observable means that its availability conveys
+information.  Making the availability of this capability conditional on client
+or server state leads to leaking information about that state.
+
```
Comment (huitema) [inline draft-thomson-train-protocol.md:465]: I have a hard time parsing the first sentence of that paragraph. Do you mean "The TRAIN is passively observable, and conveys information."? Also, not sure about "availabity. Do you mean "If the signal is only sent during specific client states, that provide information about those states" ?

**Review by huitema — COMMENTED** [review]
The unconditional part is useful. I am still concerned that the discussion of "client privacy" and "server privacy" is a bit optimistic.

**Comment by martinthomson** [comment]
Closing in favour of #27.

**Pull request closed without merging**